nixos/modules/services/monitoring/beszel-agent.nix

```diff
@@ -0,0 +1,89 @@
+{
+  lib,
+  config,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.beszel.agent;
+in
+{
+  meta.maintainers = with lib.maintainers; [
+    BonusPlay
+    arunoruto
+  ];
+
+  options.services.beszel.agent = {
+    enable = lib.mkEnableOption "beszel agent";
+
+    package = lib.mkPackageOption pkgs "beszel" { };
+
+    environment = lib.mkOption {
+      type = lib.types.attrs;
+      default = { };
+      description = ''
+        Environment variables for configuring the beszel-agent service.
+        This field will end up public in /nix/store, for secret values use `environmentFile`.
+      '';
+    };
+    environmentFile = lib.mkOption {
+      type = lib.types.nullOr lib.types.path;
+      default = null;
+      description = ''
+        File path containing environment variables for configuring the beszel-agent service in the format of an EnvironmentFile. See {manpage}`systemd.exec(5)`.
+      '';
+    };
+    extraPath = lib.mkOption {
+      type = lib.types.listOf lib.types.package;
+      default = [ ];
+      description = ''
+        Extra packages to add to beszel path (such as nvidia-smi or rocm-smi).
```

Comment:
Maybe give an example how to add nvidia-smi and rocm-smi. Before starting to use this service implementation, I added `/run/current-system/sw` to extra-path (which worked). No I had to provide a package/derivation?, so I had to add the following item to the list
`(lib.getBin config.boot.kernelPackages.nvidiaPackages.stable)`

Reply:
I think we could check for nvidia/amd GPU packages being enabled and pre-emptively  add them to the path.

Reply:
Thats one way, check if `amdgpu` or `nvidia` are inside `config.services.xserver.videoDrivers`, and add the appropriate package.
The other option is to add the `/run/current-system/sw` path to `path`:
```nix
path = [ "/run/current-system/sw" ] ++ cfg.extraPath;
```
But I am not sure if this is desired, even tho it worked in my previous setup...